cartography/intel/aws/ec2/tgw.py

```diff
@@ -0,0 +1,214 @@
+import logging
+
+from .util import get_botocore_config
+from cartography.util import run_cleanup_job
+from cartography.util import timeit
+
+logger = logging.getLogger(__name__)
+
+
+@timeit
+def get_transit_gateways(boto3_session, region):
+    client = boto3_session.client('ec2', region_name=region, config=get_botocore_config())
+    return client.describe_transit_gateways()
+
+
+@timeit
+def get_tgw_attachments(boto3_session, region):
+    client = boto3_session.client('ec2', region_name=region, config=get_botocore_config())
+    paginator = client.get_paginator('describe_transit_gateway_attachments')
+    tgw_attachments = []
+    for page in paginator.paginate():
+        tgw_attachments.extend(page['TransitGatewayAttachments'])
+    return {"TransitGatewayAttachments": tgw_attachments}
+
+
+@timeit
+def get_tgw_vpc_attachments(boto3_session, region):
+    client = boto3_session.client('ec2', region_name=region, config=get_botocore_config())
+    paginator = client.get_paginator('describe_transit_gateway_vpc_attachments')
+    tgw_vpc_attachments = []
+    for page in paginator.paginate():
+        tgw_vpc_attachments.extend(page['TransitGatewayVpcAttachments'])
+    return {"TransitGatewayVpcAttachments": tgw_vpc_attachments}
+
+
+@timeit
+def load_transit_gateways(neo4j_session, data, region, current_aws_account_id, aws_update_tag):
+    ingest_transit_gateway = """
+    MERGE (ownerAccount:AWSAccount {id: {OwnerId}})
+    ON CREATE SET ownerAccount.firstseen = timestamp(), ownerAccount.foreign = true
+    SET ownerAccount.lastupdated = {aws_update_tag}
+
+    MERGE (tgw:AWSTransitGateway {id: {TgwId}, arn: {ARN}})
```

Comment:
re: global uniqueness, not sure about Transit Gateway IDs but ARNs are globally unique, so I guess merging on the ARN alone is fine.

Reply:
I agree. we have a general to merge on what will be come the internal id in the graph (arn) in this case. using that approach, you would have to created specific attribute for the transit gateway id from aws (ie twgid) and update documentation. It's a cautious to avoid collisions which are hard to find in large graph.